plugins/chat/app/queries/chat/channel_unreads_query.rb

```diff
@@ -46,7 +46,6 @@ def self.call(channel_ids:, user_id:, include_missing_memberships: false, includ
           AND notifications.notification_type = :notification_type_mention
           AND (data::json->>'chat_message_id')::bigint > COALESCE(user_chat_channel_memberships.last_read_message_id, 0)
           AND (data::json->>'chat_channel_id')::bigint = memberships.chat_channel_id
-          AND (chat_messages.thread_id IS NULL OR chat_messages.id = chat_threads.original_message_id)
```

Comment:
This allows us to include mentions that are within threads as part of the channel mention count.